cfme/base/__init__.py

```diff
@@ -105,6 +105,29 @@ def update_advanced_settings(self, settings_dict):
         )
         assert result.ok
 
+    def upload_custom_logo(self, file_type, file_data=None, enable=True):
+        """
+        This function can be used to upload custom logo or text and use them.
+
+        Args:
+            file_type (str) : Can be either of [logo, login_logo, brand, favicon, logintext]
+            file_data (str) : Text data if file_type is logintext else image name to be uploaded
+            enable (bool) : True to use the custom logo/text else False
+        """
+        view = navigate_to(self, "CustomLogos")
+        logo_view = getattr(view.customlogos, file_type)
+        if file_data:
+            if file_type == "logintext":
+                logo_view.fill({"login_text": file_data})
+            else:
+                logo_view.fill({"image": file_data})
+                logo_view.upload_button.click()
+            view.flash.assert_no_error()
+        if logo_view.enable.is_displayed:
```

Comment:
Q: perform `enable` operation only if `is_displayed`?

Reply:
Yes, if there is no custom logo present, this button will not be visible. We want to use to this function to upload custom logo and `enable` helps us decide if we want to use that logo. If a logo is already uploaded and used, but we do not want to use it anymore, we would use this function to disable the use by passing `enable=False`.